### ISSUE TITLE
FALSE POSITIVE: `NP_ALWAYS_NULL` ignores known primitive integer values

### ISSUE THREAD

**Issue by lynae99** [spotbugs/spotbugs#4272]
Hi, I found a false positive in SpotBugs 4.10.3 when two primitive integer values are known to be equal, making a null dereference unreachable.

**Affected tool**

`SpotBugs 4.10.3`

**Affected checker**

`SpotBugs NP_ALWAYS_NULL`

**Minimal reproducer**

```java
class SpotBugsKnownIntegerValues {
  void test() {
    int left = 42;
    int right = 42;
    if (left != right) {
      Object object = null;
      object.toString();
    }
  }
}
```

**Reproduction command**

```bash
spotbugs -version
javac -d classes spotbugs-np-always-null-fp-known-integer-values.java
spotbugs -textui -effort:max -low -visitors FindNullDeref classes
```

**Current behavior**

SpotBugs reports `NP_ALWAYS_NULL` at line 7.

**Expected behavior**

SpotBugs should not report `NP_ALWAYS_NULL` because `left` and `right` are both 42, so the `left != right` branch is unreachable.
